### PR TITLE
Update CiteSeer to work from PDF paper view

### PR DESCRIPTION
**Pull request by GuyAglionby**
As the title says, as the DOI is available in the PDF url we can get the URL for the summary page and grab the information from there.

Thanks and any comments appreciated.

### CODE REVIEW

File: CiteSeer.js [hunk 1]
```diff
@@ -40,7 +40,8 @@ function detectWeb(doc, url) {
 	if ((url.indexOf('/search') != -1 || url.indexOf('/showciting') != -1) && getSearchResults(doc).length) {
 		return "multiple";
 	}
-	if (url.indexOf('/viewdoc/') != -1 && doc.getElementById('bibtex')) {
+	if ((url.indexOf('/viewdoc/') != -1 && doc.getElementById('bibtex'))
+		|| url.indexOf('/download?doi=') != -1) {
```
Comment (zuphilip) [inline CiteSeer.js]: We prefer nowadays the function `.includes()` for such tests. Can you change these two (and possibly also the ones in the line above)?

File: CiteSeer.js [hunk 2]
```diff
@@ -66,6 +67,11 @@ function doWeb(doc, url) {
 			}
 			ZU.processDocuments(articles, scrape);
 		});
+	} else if (url.indexOf('/download?doi=') !== -1) {
+		let doi = url.replace('http://citeseerx.ist.psu.edu/viewdoc/download?doi=', '');
+		doi = doi.replace('&rep=rep1', '').replace('&type=pdf', '');
```
Comment (zuphilip) [inline CiteSeer.js]: This looks a little fragile. Thus, I would prefer to work here with a regular expression, i.e. something like:
```js
let doi = url.match(/\/download\?doi=([^&]*)/);
let paperUrl = 'http://citeseerx.ist.psu.edu/viewdoc/summary?doi=' + doi[1];
ZU.processDocuments(paperUrl, scrape);
```
(Note, this code is not tested.)

**Comment by GuyAglionby** [comment]
Thanks for the comments! I added a test with the correct info, however when I run it I get an error:

```
TranslatorTester: Translating http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.415.9750&rep=rep1&type=pdf

TranslatorTester: CiteSeer Test 1: failed (Translation failed to initialize: TypeError: doc is null)
``` 

I assume this is because it's a direct link to a PDF. I looked at the translator for arXiv.org and it has a PDF test case that also fails with the same error. I'm on Zotero 5.0.34 and Scaffold 3.3.0. Let me know if you want me to push the test case anyway.

I have made the other changes as requested and tested them :)

**Comment by zuphilip** [comment]
Thank you! You are right, Scaffold tries to open the pdf directly and therefore a detection is not possible. Thus, I mentioned the test case in the comment and add some other cleanup and pushed directly to your branch.

**Pull request merged**